src/hoc/withTheme.js

```diff
@@ -23,25 +22,32 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
     static styledComponentId = 'withTheme'
 
     static contextTypes = {
-      [CHANNEL]: PropTypes.func,
+      [CHANNEL]: CONTEXT_CHANNEL_SHAPE,
```

Comment:
Yeah, let's warn that we've changed some internals that'll break third-party libraries eventually and that they should upgrade, that sounds perfect.